docs/application-tutorial/xor_distance.md

```diff
@@ -0,0 +1,94 @@
+# Xor Distance
+
+We describe how to compute a XOR distance (as known as an Hamming weight distance) in Concrete. This
+can be useful in particular for biometry use-cases, where obviously, private is a very interesting
+feature.
+
+The full code can be done [here](../../frontends/concrete-python/examples/xor_distance/hamming_distance.py). Execution times of the different functions are given in the
+final section.
+
+## The Goal
+
+Goal here is to compute
+
+```
+def dist_in_clear(x, y):
+
+    return numpy.sum(hw(x ^ y))
+```
+
+function in FHE. This function XOR to values, and compute the number of set bits of the result.
+This is a distance function, which can be used for various purpose, including measuring how two
+vectors are close to each other. In the context of biometry (or others), it may be very interesting
+to compute this function over encrypted `x` and `y` vectors.
+
+## First Implementation
+
+In the [full code](../../frontends/concrete-python/examples/xor_distance/hamming_distance.py), we use a first implementation, which is
+
+```
+def dist_in_fhe_directly_from_cp(x, y):
+    return np.sum(hw(x ^ y))
+```
+
+Here, it's a pure copy of the code in Concrete, and it compiles directly into FHE code!
+
+## Second Implementation with `fhe.bits`
+
+In the [full code](../../frontends/concrete-python/examples/xor_distance/hamming_distance.py), we use a second implementation, which is
+
+```
+def dist_in_fhe_with_bits_1b(x, y):
+    z = x + y
+    zx = fhe.bits(z)[0]
+    return np.sum(zx)
+```
+
+This function only works for bit-vectors `x` and `y` (as opposed to other functions). Here, we use
+`fhe.bits` operator to extract the least-significant bit of the addition `x+y`: indeed, this least
+signification bit is exactly `x ^ y`.
+
+## Third Implementation with concatenation
+
+In the [full code](../../frontends/concrete-python/examples/xor_distance/hamming_distance.py), we use a third implementation, which is
+
+```
+def dist_with_xor_internal(x, y, bitsize_w):
+    power = 2**bitsize_w
+    table = fhe.LookupTable([hw((i % power) ^ (i // power)) for i in range(power**2)])
+
+    z = x + power * y
+    zx = table[z]
+
+    return np.sum(zx)
```

Comment:
This can be:
```
return np.sum(fhe.multivariate(lambda x, y: hw(x ^ y))(x, y))
```
and maybe with `multivariate_strategy_preference=fhe.MultivariateStrategy.PROMOTED` configuration.

Reply:
this is done in the fourth function `dist_in_fhe_with_multivariate_internal`, but you're right: I added recently this function in the .py, but missed to add it in the .md. Doing it now

Reply:
added the multivariate_strategy_preference=fhe.MultivariateStrategy.PROMOTED flag, and now indeed, `    dist_in_fhe_with_multivariate_tables on 2 bits:  0.34 secunds` is almost as fast as `             dist_in_fhe_with_xor_tables on 2 bits:  0.32 secunds` and is in the top-2